File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String a, String b) {
        HashMap<Character,Integer> freq = new HashMap<>();

        if (a.length() != b.length()) return false;
        
        for (int i = 0; i < a.length(); i++) {
            char lowercaseA = Character.toLowerCase(a.charAt(i));
            freq.put(lowercaseA, freq.getOrDefault(a.charAt(i), 0) + 1);
            char lowercaseB = Character.toLowerCase(b.charAt(i));
            freq.put(lowercaseB, freq.getOrDefault(b.charAt(i), 0) - 1);
        }
         
        for (char c : freq.keySet()) {
            if (freq.get(c) != 0) {
                return false;
            }
        }

        return true;

    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
      if len(s) != len(t):
        return False

      freq = collections.Counter()
      for c in s:
        freq[c] += 1
      
      for c in t:
        if freq[c] == 0:
          return False
        freq[c] -= 1

      return True
      

